Monika After Story/game/script-greetings.rpy

```diff
@@ -6,37 +6,36 @@
 default persistent._mas_you_chr = False
 
 init python:
-    greetings_list=[]
-    label_list=renpy.get_all_labels()
-    for label in label_list:
-        if label.startswith('greeting_') and not renpy.seen_label(label):
-            greetings_list.append(label)
-
-    #If the greeting's list is empty, remake it without removing seen.
-    if greetings_list == []:
-        for label in label_list:
-            if label.startswith('greeting_'):
-                greetings_list.append(label)
-
+    addEvent(Event(persistent.greeting_database,eventlabel="greeting_sweetheart", random=True),eventdb=evhand.greeting_database)
 
 label greeting_sweetheart:
     m 1k "Hello again, sweetheart!"
     m 1l "It's kind of embarrassing to say out loud, isn't it?"
     m 3b "Still, I think it's okay to be embarrassed every now and then."
     return
 
+init python:
+    addEvent(Event(persistent.greeting_database,eventlabel="greeting_honey", random=True),eventdb=evhand.greeting_database)
+
 label greeting_honey:
     m 1b "Welcome back, honey!"
     m 1a "I'm so happy to see you again."
     m "Let's spend some more time together, okay?"
     return
 
+init python:
+    addEvent(Event(persistent.greeting_database,eventlabel="greeting_back", random=True),eventdb=evhand.greeting_database)
+
 label greeting_back:
     m 1a "[player], you're back!"
     m 1e "I was starting to miss you."
     m 1k "Let's have another lovely day together, alright?"
     return
 
+init python:
+    rules = {EV_RULE_GREET_RANDOM : MASGreetingRule.create_rule(skip_visual=False, random_chance=10)}
```

Comment:
Yeah so I don't think this is a good way to do this. It would be hard for regular users to add rules since they'd have to knowingly match the correct key to the correct rule. What we should be doing here would be like:
```python
    ev = Event(.......)
    MASGreetingRule.create_rule(ev, ......)
    addEvent(ev,....)
    del ev
```
**This only applies to greetings with Rules**

Reply:
OKAY Ignore what i wrote above.
If the `create_rules` functions are built in the way i described in the comment below, then the `rules` dict could be built on its own like you did originally and then set as `rules` during `Event` creation.
something like:
```python
rules = dict()
rules.update(MAS....create_rule(....))
rules.update(MAS....create_rule(...))
addEvent(Event(....rules=rules)....)
del rules
```

Reply:
I definitely love this idea, it takes away the complexity of using constants everywhere to add just a dict, while still leaving the rules not so tied to `Event`, meaning we could use those for other things in the future.